src/App.js

```diff
@@ -1,19 +1,226 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+function getTodosUserId(userId) {
+  return users.find(user => user.id === userId);
 }
 
+const todosWithUserFromFile = todos.map(todo => ({
+  ...todo,
+  user: getTodosUserId(todo.userId),
+}));
+
+export class App extends React.Component {
+  state = {
+    todosWithUser: todosWithUserFromFile,
+    todoName: '',
+    userName: '',
+    countId: todosWithUserFromFile.length,
+    noTitle: false,
+    noSelect: false,
+    maxLength: 30,
+  }
+
+  addOneId = () => {
+    this.setState(state => ({
+      countId: state.countId + 1,
+    }));
+  };
+
+  render() {
+    const {
+      todosWithUser,
+      todoName,
+      userName,
+      countId,
+      noTitle,
+      noSelect,
+      maxLength,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <form
+          action="/api/todos"
+          method="POST"
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            if (!todoName && !userName) {
+              this.setState({
+                noTitle: true,
+                noSelect: true,
+              });
+
+              return;
+            }
+
+            if (!todoName) {
+              this.setState({
+                noTitle: true,
+              });
+
+              return;
+            }
+
+            if (!userName) {
+              this.setState({
+                noSelect: true,
+              });
+
+              return;
+            }
+
+            this.setState((prevState) => {
+              const newTodo = {
+                id: countId,
+                userId: prevState.userName,
+                title: prevState.todoName
+                  .split('')
+                  .slice(0, maxLength)
+                  .join('')
+                  .replace(/[^ a-zA-Z]/g, ''),
+                completed: false,
+                user: getTodosUserId(prevState.userName),
```

Comment:
There's no need to pas the whole 'user' object, just it's name

Reply:
Small update to my previous comment:
Actually, we don't need to pass the 'name' (only 'userId').
Each TODO in our list looks like this:

```
 {
    userId: 1,
    id: 1,
    title: 'delectus aut autem',
    completed: false,
  },
```
So we should pass 'userId' to our new TODO-object, as task description says:

each TODO should have a title and userId selected from a list of given users

Also, if you want you may pass the 'Name' of the user to the TodoList to make it looks prettier, but not to add it to the new 'todo' object and it's not important for this task